src/ConnectedMode/Migration/IMigrationPrompt.cs

```diff
@@ -33,8 +42,62 @@ public interface IMigrationPrompt
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class MigrationPrompt : IMigrationPrompt
     {
-        public void Show() { }
+        private readonly INotificationService notificationService;
 
-        public void Clear() { }
+        private readonly IServiceProvider serviceProvider;
+
+        private readonly IThreadHandling threadHandling;
+
+        private const string idPrefix = "migration_";
+
+        [ImportingConstructor]
+        public MigrationPrompt([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, INotificationService notificationService, IThreadHandling threadHandling)
+        {
+            this.notificationService = notificationService;
+            this.serviceProvider = serviceProvider;
+            this.threadHandling = threadHandling;
+        }
+
+        public async Task ShowAsync()
+        {
+           await threadHandling.RunOnUIThread(() =>
```

Comment:
I am thinking now that maybe the runOnUiThread should be done inside of the GetSolutionPath() and we could just `await GetSolutionPath()`

Mostly it would be clearer which method actually relies on being on the main thread. WDYT

Reply:
Yep, sounds good.